Add blog templates

maya opened the pull request:
This adds the blog index, single blog post (large, small, and no feature image), and tag index page.

More features and functionality to come around the blog...

maya committed:
Add 10 posts and images

maya committed:
Add single blog post template

maya committed:
Style single post

maya committed:
Fix jekyll error and style tags

maya committed:
Style byline

maya committed:
Style images in blog posts

maya committed:
Create dark blog header section

maya committed:
Better style search in blog page

maya committed:
Style blog index

maya committed:
Add more sample posts

maya committed:
update current class

maya committed:
Center img on post page if less than full width

maya committed:
Fix blog link

maya committed:
Style arrow buttons

maya committed:
Add missing image

maya committed:
update current link

maya committed:
Style single post pagination

maya committed:
Add social bar

maya committed:
Add home recent posts

maya committed:
Try another way to make the class show

maya committed:
Change label to inline block so it work break

maya committed:
Change label to inline block so it work break - take 2

maya committed:
Style tags

maya committed:
Change date month to spell out fully

maya committed:
Add feature image

maya committed:
Update feature image size at mobile

maya committed:
Make blog index images link to post

maya committed:
Add number available next/previous page

maya committed:
Update blog links

maya committed:
Add one post

maya committed:
Move most inline css out

maya committed:
Add second small feature image blog post style, and no image

maya committed:
Add full month date

maya committed:
Add tags pages

maya committed:
Add search

maya committed:
Add header styles

maya committed:
Add code sample styles

maya committed:
Update previos/next post links

maya committed:
Add author to blog index posts

maya committed:
Update current link for blog posts

maya committed:
Update navigation link padding

maya committed:
Add margin to figcaption

maya committed:
Update link hover color

maya committed:
Fix post commas

maya committed:
Update tags page

maya committed:
Add border top to tag results first item

maya committed:
Update site title meta

maya committed:
Add social icons to social bar

maya committed:
Fix fonts

maya committed:
Reorder font includes

maya committed:
Update margins at mobile

maya committed:
Add html lang

maya committed:
Add #main-content id

maya committed:
Add alt text to blog post image link

maya committed:
update border accent in nav position

maya committed:
Update deliver section clears

maya committed:
Add links to social media

maya committed:
Try adding in baseurl to site search path

maya committed:
Update authors

maya committed:
Remove search js bc it didn't work

maya committed:
Make inline style more inline-y, a la @bren

Commit:
Adds nicole-fenton to authors

For some reason we never added Nicole

maya committed:
Add blog author to pages

maya committed:
Remove extra line breaks

maya committed:
Move SCSS to component files

maya committed:
Consolidate mixins

maya committed:
Tighten up if statement

maya committed:
Remove != null from if statement bc it's assumed

maya committed:
Add updated links

maya committed:
Add author, date, and keyword meta tags to HEAD

maya committed:
Update visited link color

maya committed:
Update link state colors

maya committed:
Update link colors

elainekamlley commented on _includes/related-posts.html:
```diff
@@ -0,0 +1,43 @@
+          <h2>Related Posts</h4>
+          <ul>
+          {% endif %}
+          <li class="usa-width-one-third">
```
`<li>` indent in one step

maya replied:
Done in 84c13c2.

elainekamlley commented on _includes/related-posts.html:
```diff
@@ -0,0 +1,43 @@
+              </span>
+            </article>
+          </li>
+          {% capture hasSimilar %}{{ hasSimilar }}*{% endcapture %}
```
Interested to know more about this. Seems like `hasSimilar` is capturing itself?

maya replied:
I have no idea! I found this code online here: http://zhangwenli.com/blog/2014/07/15/jekyll-related-posts-without-plugin/ and it worked.

elainekamlley commented on _layouts/post.html:
```diff
@@ -0,0 +1,90 @@
+    <header class="post-header">
+      <h1 class="post-title" itemprop="name headline">{{ page.title }}</h1>
+      <p class="post-byline">
+      {% if page.authors %}
```
Noticed that `authors` is in two places. Should we make it an `_include`?

maya replied:
Made it into an include in 1e47713. However, I don't know how to make it so it shows `{% if page.authors %}` and `{% if post.authors %}` for the different pages. So post.html uses `{% if page.authors %}`, while the tag page and blog index page uses `{% if post.authors %}`. Any idea how to do this?

elainekamlley replied:
Hmmm, @gboone any thoughts ?

gboone replied:
I feel like the `page` variable should be available for a single blog post. The `post` variable would be available in an archive view. I might have missed some context but if you want to use the same include in both situations, I think you can sniff that out with the `include.context` variable. Something like: `{% if include.context == "page" && page.authors %}` might work. Let me see if I can find some docs on that.

gboone replied:
Not docs, but here's an example of us doing it on 18f.gsa.gov. https://github.com/18F/18f.gsa.gov/blob/staging/_includes/byline.html#L1

maya replied:
@gboone if you know how to do this, would you mind making the change on this branch?

maya replied:
Fixed in 375c80a. Thanks @gboone! 

cc: @elainekamlley

gboone replied:
![](http://vignette4.wikia.nocookie.net/fantendo/images/1/1a/Troy_and_abed_handshake.gif/revision/latest?cb=20120618043638)

elainekamlley commented on _layouts/tag_index.html:
```diff
@@ -0,0 +1,59 @@
+              <a class="post-link" href="{{ post.url | prepend: site.baseurl }}">{{ post.title }}</a>
+            </h2>
+            <p class="post-byline">
+            {% if post.authors %}
```
Same question as above about `authors`

elainekamlley commented on _layouts/tag_page.html:
```diff
@@ -0,0 +1,43 @@
+---
```
What is the difference between `tag_index` and `tag_page`?

maya replied:
1. **Tag index** is a page with all the matching tag results = https://federalist.18f.gov/preview/18F/beta.18f.gov/add-blog/tags/agency-work/
2. **Tag page** is a list of all the tags = https://federalist.18f.gov/preview/18F/beta.18f.gov/add-blog/tags/

Maybe those titles don't make sense. Maybe it should be tag results for `#1` and tag index for `#2`?

elainekamlley replied:
Ohh, yea. I like that naming better.

maya replied:
Done in 09a6397.

elainekamlley commented on _sass/_components/blog-index.scss:
```diff
@@ -0,0 +1,111 @@
+// Blog index
```
Could some of these break down to smaller components? Like RSS feed for example?

maya replied:
Yes, RSS link moved to buttons.scss in c134cab, post list and pagination made into their own SCSS files in 3770842.

elainekamlley replied:
👍

elainekamlley commented on _sass/_components/blog-post.scss:
```diff
@@ -0,0 +1,197 @@
+// Blog post
```
Similar comment about `blog_index`. Could we break it into smaller components? Could help to reduce over nesting.

maya replied:
Done in 85af45a 9ae9a19 ea7e14b fa9b3ef.

elainekamlley replied:
Woo!

elainekamlley commented on _sass/_components/blog-post.scss:
```diff
@@ -0,0 +1,197 @@
+
+      img {
+        @include margin(0 null);
+      }
```
For example - this `img` tag is nested four deep.

maya replied:
It's nested in the page `.single-post` inside `article` - I think these are both necessary for this style.

elainekamlley replied:
Ah, okay. I see that now.

elainekamlley commented on _sass/_components/blog-post.scss:
```diff
@@ -0,0 +1,197 @@
+  }
+}
+
+.social_bar {
```
This could be its own component.

maya replied:
Social bar component removed after discussing with design team.

elainekamlley commented on _sass/_components/layout.scss:
```diff
@@ -48,11 +48,51 @@
+  .usa-button-primary,
+  .usa-button:visited,
+  .usa-button-primary:visited,
+  button,
```
Should this be in `buttons.scss`?

maya replied:
It's a button style but for `.background-dark` layout type, so I'm not sure if it should be moved. It's nice to see all the changes for that layout together.

msecret replied:
Would this change once we change the imports so we can override the USWDS variables?

maya replied:
No, this is a separate thing from that issue - this cascades light link and button styles when you're using a dark background.

elainekamlley replied:
Ah, okay. I'm good with this then. @msecret any other thoughts?

elainekamlley commented on _sass/_components/post-feature.scss:
```diff
@@ -12,19 +16,14 @@
     }
 
-    h3 {
+    .posts_feature-heading {
```
Does this need to be nested in `.post_feature`?

maya replied:
It does not. Fixed in 6b3dbb7.

elainekamlley commented on blog/index.html:
```diff
@@ -1,9 +1,97 @@
+            <a class="post-link" href="{{ post.url | prepend: site.baseurl }}">{{ post.title }}</a>
+          </h2>
+          <p class="post-byline">
+          {% if post.authors %}
```
Author _include?

maya replied:
Done in 1e47713.

maya committed:
Update search results template and get it working :)))

maya committed:
Add header border to search results

maya committed:
Remove social bar

maya committed:
Style search results button

maya committed:
Remove blank space

maya committed:
Indent li one level

maya committed:
Add post author as include

maya committed:
Use mixin for RSS link

maya committed:
Move RSS to buttons

maya committed:
Move pagination and post list to separate SCSS files

maya committed:
Fix spacing

maya committed:
De-nest post feature heading

maya committed:
Move out tags to its own scss file and remove uneeded scss

maya committed:
Move header border styles to layouts

maya committed:
Move scss to component files

maya committed:
Move back to blog button to buttons.scss

maya committed:
Fix search so it doesn't duplicate results

maya committed:
Fix search baseurl issue and remove blog index results

maya committed:
Update tag page names

Commit:
Updates jekyll and other gems and adds some include logic

Updates jekyll to be even with 18f.gsa.gov, fixing some bugs in rendering and adding some performance improvements.

Adds a "context" variable to the post-author include so that we can use
the same code from within a list of posts *or* a single page.

Commit:
Fixes the archive layout

Commit:
fixup! Fixes the archive layout

elainekamlley commented:
Awesome work @maya and @gboone! Merging!
